### ISSUE TITLE
User enumeration at login

### ISSUE THREAD

**Issue by xTibor** [redox-os/userutils#34]
The `login` program is vulnerable to user enumeration. It only asks for a password and does a timeout when the specified user account exists, thus allowing malicious actors to brute force the possible users at a rate of hundreds of usernames per second. (when login asks for a password = valid user on the system).

**Demo:**
https://www.youtube.com/watch?v=7XfipgWmpxM

**Possible fix:**
Ask for a password and do a timeout even the specified user doesn't exist.

**Comment by MggMuggins** [comment]
I would propose a new function in redox_users, `timeout`, which simply sleeps for the configured timeout. `login` can then call this function if a user does not exist.

**Comment by xTibor** [comment]
I don't think that fully solves the problem. Sleeping without a password prompt on invalid users still allows people to determine the existence of the users on the system. It just slows things down. The login prompt must behave exactly the same for valid and invalid users.
